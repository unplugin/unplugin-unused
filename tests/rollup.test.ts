import path from 'node:path'
import process from 'node:process'
import {
  rollupBuild,
  RollupToStringPlugin,
  testFixtures,
} from '@sxzz/test-utils'
import { describe } from 'vitest'
import UnpluginUnused from '../src/rollup.ts'

describe('rollup', async () => {
  const { dirname } = import.meta
  await testFixtures(
    '*.{js,tsx,vue}',
    async (args, id) => {
      try {
        const { snapshot } = await rollupBuild(id, [
          id.includes('early-error')
            ? {
                name: 'rollup-plugin-early-error',
                transform() {
                  this.error(new Error('Early error'))
                },
              }
            : false,
          UnpluginUnused({
            level: 'error',
          }),
          RollupToStringPlugin(),
        ])
        return snapshot
      } catch (error: any) {
        error.message = error.message.replaceAll(
          process.cwd() + path.sep,
          '#CWD#/',
        )
        throw error
      }
    },
    {
      cwd: path.resolve(dirname, 'fixtures'),
      promise: true,
    },
  )
})
