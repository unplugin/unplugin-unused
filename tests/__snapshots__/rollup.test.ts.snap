// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rollup > basic.js 1`] = `
"// basic.js
var basic = \`import '@rollup/pluginutils'
import 'js-tokens'
import 'pkg-types'
import 'unplugin'
import 'picocolors'
\`;

export { basic as default };
"
`;

exports[`rollup > error.js 1`] = `
[Error: Unused [36m2[39m dependencies found: 

[33m-[39m [1mjs-tokens[22m
[33m-[39m [1mpicocolors[22m

You can remove them from #CWD#/package.json]
`;

exports[`rollup > error-comment.js 1`] = `
[Error: Unused [36m4[39m dependencies found: 

[33m-[39m [1m@rollup/pluginutils[22m
[33m-[39m [1mjs-tokens[22m
[33m-[39m [1mpicocolors[22m
[33m-[39m [1mpkg-types[22m

You can remove them from #CWD#/package.json]
`;

exports[`rollup > error-tsx.tsx 1`] = `
[Error: Unused [36m2[39m dependencies found: 

[33m-[39m [1mpicocolors[22m
[33m-[39m [1munplugin[22m

You can remove them from #CWD#/package.json]
`;

exports[`rollup > error-vue.vue 1`] = `
[Error: Unused [36m4[39m dependencies found: 

[33m-[39m [1m@rollup/pluginutils[22m
[33m-[39m [1mjs-tokens[22m
[33m-[39m [1mpicocolors[22m
[33m-[39m [1mpkg-types[22m

You can remove them from #CWD#/package.json]
`;
